update.go

```diff
@@ -254,6 +259,18 @@ other options are ignored.
 		config.Cgroups.Resources.MemorySwap = *r.Memory.Swap
 		config.Cgroups.Resources.PidsLimit = r.Pids.Limit
 
+		if intelrdt.IsEnabled() && config.IntelRdt != nil {
```

Comment:
Wouldn't it be better to just set the value and let the libcontainer rtd code validate and return an error if its not enabled?  Right now, it would just fail silently and no one would know if its set or not

Reply:
@crosbymichael 

I have made changes according to your suggestion to avoid silent failure.
There are several cases in error paths:
(1) IsEnabled == false && intelRdt is specified in config
(2) IsEnabled == true && intelRdt is not specified in config
(3) IsEnabled == true && intelRdt is specified in config && update value is invalid

(1): It is handled by validate.Validator(). runc create/run will fail. It never pass to update command.
(2): We'd better return an error that the update command is failed before container.Set(). Just passing through container.Set() can't handle this case:
(3): We just set the value, intelrdt.Set() will return an error that the update value is invalid.